doc: Add a note about dependencies for `Bootloader=systemd-boot` (#3482)
systemd/mkosi: mkosi/resources/man/mkosi.1.md

    in the ESP for compatibility with signed versions of grub which load
    the grub configuration from this location.

>     Note: When using `systemd-boot` or `systemd-boot-signed`, the image

Reviewer: Since this is more of a footnote type thing and more paragraphs follow, could you move this at the end of the option description, i.e. after the paragraph that starts with "Kernels need to be placed"?

>     must include `bootctl`. This usually requires adding a package, e.g.
>     `systemd-boot` on Debian-based distributions or `systemd` on Arch.

Reviewer: Isn't it hard not to have `systemd` on Arch? How about
```suggestion
    must include `bootctl`. This may require adding a package when it is packaged separately, e.g.
    `systemd-boot` on Debian-based distributions.
```
